src/app/sitemap.ts

```diff
@@ -0,0 +1,111 @@
+import type { MetadataRoute } from "next";
+import type { CategoryType } from "@/lib/types/categories";
+
+export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
+  const apiURL = process.env.GRAPHQL_API_URL || "";
+  const baseURL = "https://volt-eosin.vercel.app";
+  const latestDate = new Date("2024-11-30T00:00:00.000Z");
+
+  //all the static URLs
+  const staticUrls: MetadataRoute.Sitemap = [
+    { url: `${baseURL}`, lastModified: latestDate },
+    { url: `${baseURL}/learn`, lastModified: latestDate },
+    { url: `${baseURL}/tools`, lastModified: latestDate },
+    { url: `${baseURL}/contact-us`, lastModified: latestDate },
+    { url: `${baseURL}/faqs`, lastModified: latestDate },
+    { url: `${baseURL}/terms-and-conditions`, lastModified: latestDate },
+    { url: `${baseURL}/privacy-policy`, lastModified: latestDate },
+    { url: `${baseURL}/about-us`, lastModified: latestDate },
+  ];
+
+  let dynamicURLS: MetadataRoute.Sitemap = [];
+
+  //fetch data for the dynamic URLs
+  try {
+    const response = await fetch(apiURL, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        query: `
+            query getData {
+              tags(where: {orderby: COUNT, order: DESC}, first: 10) {
+                nodes {
+                  slug
+                }
+              }
+              categories {
+                nodes {
+                  slug
+                }
+              }
+              posts(first: 60) {
+                nodes {
+                  slug
+                  modified
+                }
+              }
+              users(where: {hasPublishedPosts: POST}) {
+                nodes {
+                  slug
```

Comment:
Create a ticket for implementing graphql codegen

Reply:
Ticket created